src/__swaps__/screens/Swap/components/ReviewPanel.tsx

```diff
@@ -1,33 +1,93 @@
 import Animated, {
   runOnJS,
+  runOnUI,
   useAnimatedReaction,
   useAnimatedStyle,
   useDerivedValue,
   useSharedValue,
   withTiming,
 } from 'react-native-reanimated';
-import { NavigationSteps, useSwapContext } from '../providers/swap-provider';
-import { fadeConfig } from '../constants';
-import { ethereumUtils } from '@/utils';
-import { ChainImage } from '@/components/coin-icon/ChainImage';
+
+import { AnimatedText, Box, Inline, Separator, Stack, Text, globalColors, useColorMode } from '@/design-system';
+import * as i18n from '@/languages';
+import { CrosschainQuote, Quote, QuoteError } from '@rainbow-me/swaps';
+import { useAccountSettings } from '@/hooks';
+
+import { NavigationSteps, useSwapContext } from '@/__swaps__/screens/Swap/providers/swap-provider';
+import { fadeConfig } from '@/__swaps__/screens/Swap/constants';
 import { ChainId } from '@/__swaps__/types/chains';
-import { chainNameFromChainIdWorklet } from '@/__swaps__/utils/chains';
-import { AnimatedSwitch } from './AnimatedSwitch';
+import { chainNameForChainIdWithMainnetSubstitutionWorklet } from '@/__swaps__/utils/chains';
+import { AnimatedSwitch } from '@/__swaps__/screens/Swap/components/AnimatedSwitch';
 import { GasButton } from '@/__swaps__/screens/Swap/components/GasButton';
-import { GestureHandlerV1Button } from './GestureHandlerV1Button';
-import { StyleSheet, View } from 'react-native';
-import { AnimatedChainImage } from './AnimatedChainImage';
+import { GestureHandlerV1Button } from '@/__swaps__/screens/Swap/components/GestureHandlerV1Button';
+import { AnimatedChainImage } from '@/__swaps__/screens/Swap/components/AnimatedChainImage';
+import { convertRawAmountToBalance, convertRawAmountToNativeDisplay, handleSignificantDecimals, multiply } from '@/__swaps__/utils/numbers';
+import { useNativeAssetForChain } from '@/__swaps__/screens/Swap/hooks/useNativeAssetForChain';
+
+const unknown = i18n.t(i18n.l.swap.unknown);
 
 const RainbowFee = () => {
+  const { nativeCurrency } = useAccountSettings();
   const { isDarkMode } = useColorMode();
+  const { quote, internalSelectedInputAsset } = useSwapContext();
+
+  const { nativeAsset } = useNativeAssetForChain({ inputAsset: internalSelectedInputAsset });
 
-  const rainbowFee = useSharedValue(i18n.t(i18n.l.swap.unknown));
+  const index = useSharedValue(0);
+  const rainbowFee = useSharedValue<string[]>([unknown, unknown]);
+
+  const feeToDisplay = useDerivedValue(() => {
+    return rainbowFee.value[index.value];
+  });
 
-  return <AnimatedText align="right" color={isDarkMode ? 'labelSecondary' : 'label'} size="15pt" weight="heavy" text={rainbowFee} />;
+  const swapIndex = () => {
+    'worklet';
+    index.value = 1 - index.value;
+  };
+
+  const calculateRainbowFeeFromQuoteData = useCallback(
+    (quote: Quote | CrosschainQuote) => {
+      const updateRainbowFee = ({ feeInEth, feePercentage }: { feeInEth: string; feePercentage: string }) => {
```

Comment:
i don't think this needs to be a worklet either. if it doesn't you can also remove this function and just set rainbowFee.value directly where this func is now called